scripts/bulkimport.sh

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env bash
+${DEBUG:+set -x}
+set -e
+set -o errexit -o pipefail -o noclobber -o nounset
+
+HOME=${HOME:-"~"}
+
+command -v curl >/dev/null || (echo "You must install curl to proceed" >&2 && exit 1)
+command -v jq >/dev/null || (echo "You must install jq to proceed" >&2 && exit 1)
+command -v terraform >/dev/null || (echo "You must install terraform to proceed" >&2 && exit 1)
+
+function usage {
+	echo "${0} [{--users |-u} {--repos|-r} | {--groups|-g}] -h|--host https://\${host}" >&2
+	echo "duplicate resource declarations are de duped (see below)"
+	echo "example: ${0} -u --repos --users -h https://myartifactory.com > import.tf"
+	exit 1
+}
+
+resources=()
+while getopts urdga:-: OPT; do
+	# shellcheck disable=SC2154
+	if [ "$OPT" = "-" ]; then # long option: reformulate OPT and OPTARG
+		OPT="${OPTARG%%=*}"      # extract long option name
+		OPTARG="${OPTARG#$OPT}"  # extract long option argument (may be empty)
+		OPTARG="${OPTARG#=}"     # if long option argument, remove assigning `=`
+	fi
+	# shellcheck disable=SC2214
+	case "${OPT}" in
+	u | users)
+		resources+=(users)
+		;;
+	r | repos)
+		resources+=(repos)
+		;;
+	g | groups)
+		resources+=(groups)
+		;;
+	a | artifactory-url)
+		host="${OPTARG}"
+		grep -qE 'http(s)?://.*' <<<"${host}" || (echo "malformed url name: ${host}. must be of the form http(s)?://.*" && exit 1)
+		;;
+	??*)
+		usage
+		;;
+	*)
+		usage
+		;;
+	esac
+done
+
+function netrc_location {
```

Comment:
All net rc negative case code is untested. But, I have to write to the `netrc` file for curl so passwords don't show up in the process anywhere